### PR TITLE
Fix: particleRenderer bug

### PR DESCRIPTION
**Pull request by eyworldwide**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/oasis-engine/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

### What kind of change does this PR introduce? (Bug fix, feature, docs update, ...)
Bug fix.

### What is the current behavior? (You can also link to an open issue here)

- particleRenderer is not hidden before start to play or after life ends. (Issue: https://github.com/oasis-engine/engine/issues/241)
- particle renderer is still updating after the end of once play.

### What is the new behavior (if this is a feature change)?
- particleRenderer is hidden before start to play or after life ends. 
- particle renderer will stop updating after the end of once play which is better for performance.

### Does this PR introduce a breaking change? (What changes might users need to make in their application due to this PR?)
No.

### Other information:
No.

### CODE REVIEW

File: packages/core/src/shaderlib/extra/particle.vs.glsl [hunk 1]
```diff
@@ -40,25 +39,15 @@ void main() {
...
-  }
-
-  // Not born means death, otherwise it will be displayed if not born
-  if (deltaTime == 0.0) {
-    deltaTime = life;
+  if ((u_once && u_time > life + startTime)) {
+    deltaTime = 0.0;
   }
 
   v_lifeLeft = 1.0 - deltaTime / a_lifeAndSize.y;
```
Comment (singlecoder) [inline packages/core/src/shaderlib/extra/particle.vs.glsl:53]: Variables like a_lifeAndSize are difficult to understand

Comment (singlecoder) [inline packages/core/src/shaderlib/extra/particle.vs.glsl:53]: v_lifeLeft = 1.0 - deltaTime / life is better

Comment (eyworldwide) [reply packages/core/src/shaderlib/extra/particle.vs.glsl]: fixed.

Comment (eyworldwide) [reply packages/core/src/shaderlib/extra/particle.vs.glsl]: fixed

**Review by singlecoder — APPROVED** [review]
+1

**Pull request merged**